xbmc/addons/interfaces/GUI/DialogFileBrowser.cpp

```diff
@@ -0,0 +1,341 @@
+                          __FUNCTION__, shares, heading, path_in, path_out, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = path_in;
+
+  VECSOURCES vecShares;
+  GetVECShares(vecShares, shares, strPath);
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetDirectory(vecShares, heading, strPath, write_only);
+  if (bRet)
+    *path_out = strdup(strPath.c_str());
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::show_and_get_file(void* kodiBase, const char* shares, const char* mask, const char* heading,
+                                                       const char* path_in, char** path_out, bool use_thumbs, bool use_file_directories)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!shares || !mask || !heading || !path_in || !path_out)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid handler data (shares='%p', mask='%p', heading='%p', path_in='%p', path_out='%p') on addon '%s'",
+                          __FUNCTION__, shares, mask, heading, path_in, path_out, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = path_in;
+
+  VECSOURCES vecShares;
+  GetVECShares(vecShares, shares, strPath);
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetFile(vecShares, mask, heading, strPath, use_thumbs, use_file_directories);
+  if (bRet)
+    *path_out = strdup(strPath.c_str());
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::show_and_get_file_from_dir(void* kodiBase, const char* directory, const char* mask,
+                                                                const char* heading, const char* path_in, char** path_out,
+                                                                bool use_thumbs, bool use_file_directories, bool single_list)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!directory || !mask || !heading || !path_in || !path_out)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIDialogFileBrowser::%s - invalid handler data (directory='%p', mask='%p', heading='%p', path_in='%p', path_out='%p') on addon '%s'",
+                          __FUNCTION__, directory, mask, heading, path_in, path_out, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = path_in;
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetFile(directory, mask, heading, strPath, use_thumbs, use_file_directories, single_list);
+  if (bRet)
+    *path_out = strdup(strPath.c_str());
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::show_and_get_file_list(void* kodiBase, const char* shares, const char* mask,
+                                                            const char* heading, char*** path, unsigned int* entries,
```

Comment:
I would name the variable file_list, not path as it contains a list of several files, not a single path.